### PR TITLE
Add API conversion interface for PReLU & BatchNormalization

### PR DESCRIPTION
**Pull request by fredtony**

### CODE REVIEW

File: keras/legacy/interfaces.py [hunk 1]
```diff
@@ -133,3 +133,64 @@ def wrapper(*args, **kwargs):
...
+        func: `__init__` method of `PReLU`.
+
+    # Returns
+        A constructor conversion wrapper.
+    """
+    @six.wraps(func)
+    def wrapper(*args, **kwargs):
+        if len(args) > 1:
+            # The first entry in `args` is `self`.
+            raise TypeError('The `PReLU` layer can have no '
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: This layer is commonly used with a positional argument, e.g. `PReLU(0.4)`. This should still work.

File: keras/legacy/interfaces.py [hunk 2]
```diff
@@ -133,3 +133,64 @@ def wrapper(*args, **kwargs):
...
+    @six.wraps(func)
+    def wrapper(*args, **kwargs):
+        if len(args) > 1:
+            # The first entry in `args` is `self`.
+            raise TypeError('The `BatchNormalization` layer can have no '
+                            'positional arguments.')
+
+        # mode argument no longer accepted
+        if 'mode' in kwargs:
+            raise TypeError('`mode` argument no longer accepted. '
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: The point of these conversion interfaces is that old code should still work. So in this case we should figure out a better solution. Please leave out this layer.

File: keras/legacy/interfaces.py (hunk 1 above)
Comment (fredtony) [inline keras/legacy/interfaces.py]: I'm not sure that I've seen PReLU used with a positional argument in that way. I know that the parameters are learned, so I'm assuming that argument must be referring to an initializer. In PReLU(0.4), does this set the initializer of the weights as uniform/constant weights of 0.4?

File: keras/legacy/interfaces.py (hunk 2 above)
Comment (fredtony) [inline keras/legacy/interfaces.py]: I'll leave it out for now. I've been looking at the backends and it seems like there might be a way to add the functionality back with some work. Can I attempt to research and find a way to change it up or would you rather leave it out?

Comment (fchollet) [inline keras/legacy/interfaces.py]: Just leave it out, we'll think of something later.

File: keras/legacy/interfaces.py (hunk 1 above)
Comment (fchollet) [inline keras/legacy/interfaces.py]: Sorry, I was confusing it with `LeakyReLU`. You're right.
